### ISSUE TITLE
Custom rule exemption not working for fields

### ISSUE THREAD

**Issue by chchtv11** [looker-open-source/look-at-me-sideways#123]
I am unable to add an exemption at the field level for custom rules. However, I can add them at the view level.

Example:

In manifest:
```
# rule: 3-d_value_format {
#  description: "Use of 'value_format' in dimension or measure. Use (and define if needed) 'value_format_name' instead"
#  match: "$.files[*].view.*.*.*"
#  expr_rule: (== ::match:value_format undefined) ;;
# }
```

In view file:

```measure: total_value {
  #  LAMS
  #  rule_exemptions: {
  #   3-d_value_format: "Whatever reason"
  #  }
    type: sum
    sql: ${value} ;;
    value_format: "0.00"
  }
```

I understand this is similar to another issue #50 but I thought it might be specific to field-level exemptions since I can add exemptions to views.

**Comment by fabio-looker** [comment]
Hi @chchtv11 

Dimension-level exemptions seem to work fine in my testing: https://github.com/looker-open-source/look-at-me-sideways/blob/issue-123/__tests__/dummy-projects/09-exempt-custom-rules/index.test.js

![image](https://user-images.githubusercontent.com/23034640/221220616-c9d2745e-b7b4-4bfd-84e3-492455594aa4.png)

Perhaps you could provide a more specific test case? and also confirm which version of LAMS you are running?

**Comment by chchtv11** [comment]
Hi @fabio-looker thank you for looking into this. I am running LAMS v2 in github actions. I have copied the test you ran into my project (but specifying measure only, as `['dimension', 'measure']` has never worked for me)

In `manifest.lkml`:
```
# rule: types_required {
#  description: "Type must be specified"
#  match: "$.model.*.view.*.measure.*"
#  expr_rule: ( !== ::match:type undefined ) ;;
# }
```

In `sales_target_team.view.lkml`:
```
  measure: total_value {
    #  LAMS
    #  rule_exemptions: {
    #   types_required: "Please be exempt"
    #  }
    sql: 1 ;;
  }
```
As you can see it is failing
<img width="905" alt="Screenshot 2023-02-27 at 16 52 34" src="https://user-images.githubusercontent.com/10041104/221612544-0d6cbcb0-ca00-4ef7-a54d-d8167cf183c5.png">
<img width="726" alt="Screenshot 2023-02-27 at 16 51 10" src="https://user-images.githubusercontent.com/10041104/221612577-974da78e-f127-4113-af11-614841064496.png">

**Comment by fabio-looker** [comment]
Thanks, I found a typo in my test and am now able to reproduce this. Actually I'm suprised no one has mentioned it before! I'll try to figure out the cause by the end of the week

**Comment by fabio-looker** [comment]
I believe the cause of this is https://github.com/looker-open-source/look-at-me-sideways/blob/master/lib/custom-rule/rule-evaluator.js#L19

`exempt: exempt || getExemption(modelFragment, ruleDef.$name),` should be `exempt: exempt || getExemption(modelFragment[pathpart], ruleDef.$name)`

I have this fixed in my v3 branch which I am hoping to release this month.

**Comment by fabio-looker** [comment]
This is currently available in the v3 beta: https://www.npmjs.com/package/@looker/look-at-me-sideways/v/3.0.0-beta.0

Release notes: https://github.com/looker-open-source/look-at-me-sideways/blob/fabble/v3-extends-refinements/docs/release-notes/v3.md

If you do not want to upgrade to v3, let me know and I can also patch this into v2

**Comment by fabio-looker** [comment]
Hello!

Although this was fixed in 3.0.0-beta.0 , that beta release had several issues. Many of those issues are fixed in the next beta pre-release [3.0.0-beta.1](https://www.npmjs.com/package/@looker/look-at-me-sideways/v/3.0.0-beta.1).

I'll keep this issue open until v3 is out of beta. Thank you for your feedback thus far!

**Comment by fabio-looker** [comment]
Hi @chchtv11 ,

[v3](https://looker-open-source.github.io/look-at-me-sideways/release-notes/v3) is now generally available. I'm closing out this ticket, but let me know if you have any feedback and I can reopen it as needed.
